packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/watch-blocks/watch-blocks-v1-endpoint.ts

```diff
@@ -0,0 +1,329 @@
+  }
+
+  return sanitizeHtml(safeStringify(error));
+}
+
+/**
+ * Endpoint to watch for new blocks on fabric ledger and report them
+ * to client using socketio.
+ */
+export class WatchBlocksV1Endpoint {
+  public static readonly CLASS_NAME = "WatchBlocksV1Endpoint";
+
+  private readonly log: Logger;
+  private readonly socket: SocketIoSocket<
+    Record<WatchBlocksV1, (next: WatchBlocksResponseV1) => void>
+  >;
+
+  constructor(public readonly config: IWatchBlocksV1EndpointConfiguration) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(config, `${fnTag} arg options`);
+    Checks.truthy(config.socket, `${fnTag} arg options.socket`);
+
+    this.socket = config.socket;
+
+    const level = this.config.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+  }
+
+  /**
+   * Get this class name.
+   */
+  public get className(): string {
+    return WatchBlocksV1Endpoint.CLASS_NAME;
+  }
+
+  /**
+   * Callback executed when receiving block with custom cactus type "cactus:transactions"
+   * Sends WatchBlocksV1.Next with new block to the client.
+   *
+   * @param blockEvent full block
+   *
+   * @returns Nothing.
+   */
+  private monitorCactusTransactionsCallback(blockEvent: BlockEvent) {
+    const { socket, log } = this;
+    const clientId = socket.id;
+    log.debug(
+      `CactusTransactions BlockEvent received: #${blockEvent.blockNumber.toString()}, client: ${clientId}`,
+    );
+
+    if (!("data" in blockEvent.blockData)) {
+      log.error("Wrong blockEvent type received - should not happen!");
+      return;
+    }
+
+    const blockData = blockEvent.blockData.data?.data as any;
+    if (!blockData) {
+      log.debug("Block data empty - ignore...");
+      return;
+    }
+
+    const transactions: WatchBlocksCactusTransactionsEventV1[] = [];
+    for (const data of blockData) {
+      try {
+        const payload = data.payload;
+        const transaction = payload.data;
+        const actionPayload = transaction.actions[0].payload;
+        const proposalPayload = actionPayload.chaincode_proposal_payload;
+        const invocationSpec = proposalPayload.input;
+
+        // Decode args and function name
+        const rawArgs = invocationSpec.chaincode_spec.input.args as Buffer[];
+        const decodedArgs = rawArgs.map((arg: Buffer) => arg.toString("utf8"));
+        const functionName = decodedArgs.shift() ?? "<unknown>";
+
+        const chaincodeId = invocationSpec.chaincode_spec.chaincode_id.name;
+        const channelHeader = payload.header.channel_header;
+        const transactionId = channelHeader.tx_id;
+
+        transactions.push({
+          chaincodeId,
+          transactionId,
+          functionName,
+          functionArgs: decodedArgs,
+        });
+      } catch (error) {
+        log.error(
+          "Could not retrieve transaction from received block. Error:",
+          safeStringifyException(error),
+        );
```

Comment:
@outSH If we only log here to the error logs of the back-end then how does the client application get notified that transactions will be missing from the response? 

If a sys-admin sees the logs and sees these errors, will they know what to do to fix it? Or if they passed those error logs to us, would we be able to do anything about it?

Reply:
@petermetz This means there's a bug in connector itself, or the connector we use is not compatible with fabric ledger we're connected to (i.e. fabric block data changed). Monitoring in fabric-socketio ignored block like that, maybe it should be more explicit in informing the client that there were some blocks it couldn't parse so that the client app could react accordingly? But should it stop entire monitoring? I will think about notifying the client

Reply:
@outSH Thank you!